pkg/hoist/hoist_launchable.go

```diff
@@ -133,19 +134,15 @@ func (hl *Launchable) InvokeBinScript(script string) (string, error) {
 		return "", err
 	}
 
-	cmd := exec.Command(
-		hl.P2exec,
-		"-n",
-		"-u",
-		hl.RunAs,
-		"-e",
-		hl.ConfigDir,
-		"-l",
-		hl.CgroupConfigName,
-		"-c",
-		hl.Id,
-		cmdPath,
-	)
+	p2ExecArgs := util.P2ExecArgs{
+		Command:          cmdPath,
+		User:             hl.RunAs,
+		EnvDir:           hl.ConfigDir,
+		NoLimits:         hl.ExecNoLimit,
+		CgroupConfigName: hl.CgroupConfigName,
+		CgroupName:       hl.Id,
+	}
+	cmd := exec.Command(hl.P2Exec.Bin, hl.P2Exec.BuildWithArgs(p2ExecArgs)...)
```

Comment:
For this use, it's unfortunate that you have to specify the first argument separately.